File: Client/src/app/model/graph.model.ts
import { WhileModel } from "./while.model";

export class GraphModel {
    public linkDataArray: LinkDataArrayModel[];
    public nodeDataArray: NodeDataArrayModel[];
    public nodeKeyProperty: string;
    public whileGroups: WhileModel[];
    public class: string;

    constructor() {
        this.whileGroups = [];
        this.linkDataArray = [];
        this.nodeDataArray = [];
    }
}

export class NodeDataArrayModel {
    public key?: number;
    public text: string;
    public color?: string;
    public loc?: string;
    public isIf?: boolean;
    public isWhile?: boolean;
    public isSwitch?: boolean;
    public isInSwitch?: boolean;
    public doNodeId?: number;
    public elseNodeId?: number;
    public childNodeId?: number;
    public isCommon: boolean;
    public id?: number;
    public description: string;
    public taskId: number;
    public isInWhile: boolean;
    public caseValue?: string;
    public isRoot: boolean;
    public isDefault: boolean;
    public pendingNumber?: number;
    public doneNumber?: number;
    public stopNumber?: number;
    public startDate?: string;
    public endDate?: string;
    public startDateDescription?: string;
    public endDateDescription?: string;
    public pendingDescription?: string;
    public doneDescription?: string;
    public stopDescription?: string;
    public doneDataModels?: DoneDataModel[];
    public pendingDataModels?: DoneDataModel[];
    public stopDataModels?: DoneDataModel[];
    public status?: string;
    public deadLine?: number;
    public deadLineDescription?: string;
}

export class DoneDataModel {
    public RequestNumber: string;
    public UniqKey: string;
}

export class LinkDataArrayModel {
    public from: number;
    public to: number;
    public text?: string;
    public curviness?: string;
    public points?: number[];
}

export class NodeDataToAdd {
    public key: number;
    public text: string;
    public color?: string;
    public loc: string;
    public isIf: boolean;
    public ifResult: boolean;
    public parentNode: NodeDataToAdd;
    public doNodeId: number;
    public elseNodeId: number;
    public childNodeId?: number;
    public description: string;
    public isCommon: boolean;
    public taskId?: number;
}

File: Server/Contract/GraphModel.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace workflow.Core.Service.Contracts
{
    
    public class GraphModel
    {
        
        public string GraphJson { get; set; }
        
        public int WorkflowId { get; set; }
        
        public long WorkflowVersion { get; set; }
        
        public List<NodeDataArray> NodeDataArray { get; set; } = new List<NodeDataArray>();
        
        public List<LinkDataArray> LinkDataArray { get; set; } = new List<LinkDataArray>();
        
        public List<WhileModel> WhileGroups { get; set; }
    }

    
    public class WhileModel
    {
        
        public List<int> TaskIds { get; set; }
        
        public int WhileKey { get; set; }
    }
    
    public class LogDataModel
    {
        
        public string RequestNumber { get; set; }
        
        public string UniqKey { get; set; }
    }

    
    public class NodeDataArray
    {
        
        public string StartDate { get; set; }
        
        public string EndDate { get; set; }
        
        public string Status { get; set; }
        
        public List<LogDataModel> DoneDataModels { get; set; }
        
        public List<LogDataModel> PendingDataModels { get; set; }
        
        public List<LogDataModel> StopDataModels { get; set; }
        
        public int Key { get; set; }
        
        public bool IsRoot { get; set; }
        
        public string Text { get; set; }
        
        public string Color { get; set; }
        
        public int TaskId { get; set; }
        
        public string Loc { get; set; }
        
        public bool IsIf { get; set; }
        
        public bool IsSwitch { get; set; }
        
        public string CaseValue { get; set; }
        
        public int DoNodeId { get; set; }
        
        public int ElseNodeId { get; set; }
        
        public bool IsInWhile { get; set; }
        
        public bool IsInSwitch { get; set; }
        
        public bool IsWhile { get; set; }
        
        public bool? IsCommon { get; set; }
        
        public bool? IsDefault { get; set; }
        
        public int PendingNumber { get; set; }
        
        public int DoneNumber { get; set; }
        
        public int StopNumber { get; set; }
        
        public int DeadLine { get; set; }
        
        public string Description { get; set; }

    }

    
    public class LinkDataArray
    {
        
        public int From { get; set; }
        
        public int To { get; set; }
        
        public string Text { get; set; }
        
        public List<float> Points { get; set; }

    }
}
